src/stack/stack.jl

```diff
@@ -29,6 +29,9 @@ abstract type AbstractDimStack{K,T,N,L} end
 const AbstractVectorDimStack = AbstractDimStack{K,T,1} where {K,T}
 const AbstractMatrixDimStack = AbstractDimStack{K,T,2} where {K,T}
 
+(::Type{T})(st::AbstractDimStack) where T<:AbstractDimArray =
+    T([st[D] for D in DimIndices(st)]; dims=dims(st), metadata=metadata(st))
```

Comment:
Does this mean that `Raster(RasterStack(...))` is actually an expensive operation / materializes everything in memory?  Is there a way we could avoid this, maybe a `ConcatDimArray` or something?

Reply:
Yeah in this implementation. `UnstackedDimArray` or `InterleavedDimArray` might be a better name because its really interleaving the layers not concatenating. 

I've also been thinking of just making `AbstractDimStack` an `AbstractArray` itself which would make that a moot point. This PR is a step in that direction.

Reply:
For now lets just make it expensive